tests/main/snapd-without-core/task.yaml

```diff
@@ -0,0 +1,46 @@
+summary: Ensure the snapd works without core on classic systems
+
+# Run only on ubuntu classic because this re-execs.  We have a
+# separate test for UC18 already and on UC16 we always have a core
+# snap so we don't need to test there.
+systems: [ubuntu-1*, ubuntu-2*]
+
+restore: |
+    rm -f /tmp/snapd_*.snap
+
+execute: |
+    if [ "${SNAP_REEXEC:-}" = "0" ]; then
+        echo "skipping test when SNAP_REEXEC is disabled"
+        exit 0
+    fi
+
+    echo "Create modified snapd snap"
+    UNPACK_DIR="./snapd-unpack"
+    snap download --edge snapd
+    unsquashfs -no-progress -d "$UNPACK_DIR" snapd_*.snap
+    dpkg-deb -x "$SPREAD_PATH"/../snapd_*.deb "$UNPACK_DIR"
+    cp /usr/lib/snapd/info "$UNPACK_DIR"/usr/lib/
+    snap pack "$UNPACK_DIR" /tmp
```

Comment:
The same logic exists in prepare.sh in setup_reflash_magic, could we have it in a common helper instead? I can imagine more tests requiring snapd snap where this will be needed.